backtracking/rat_in_maze.py

```diff
@@ -0,0 +1,104 @@
+def print_matrix(matrix):
+    """
+    This method takes matrix as parameter and print the solution
+    In this matrix 1 s are the path to the destination.
+    Parameters :
+        matrix(2D matrix) : matrix
+    """
+    for i in matrix:
+        print(i)
+
+
+def fill_matrix(matrix, i, j):
+    """
+    This method just fills matrix i, j with value
+    Parameters :
+        matrix(2D matrix) : matrix
+        i, j : coordinates of the matrix.
+    """
+    matrix[i][j] = 1
+
+
+def solveMaze(maze, size):
+    """
+    This method solves rat in maze algorithm.
+    In this problem we have n by n matrix and we have start point and end point
+    we want to go from source to distination. In this matrix 0 are block paths
+    1 are open paths we can use.
+    Parameters :
+        maze(2D matrix) : maze
+        size : size of our maze(square)
+    Returns:
+        This method returns true and array if solution found otherwise false and None
+    """
+    # We need to create solution object to save path.
+    solutions = [[0 for _ in range(size)] for _ in range(size)]
+    flag = runmaze(maze, size, 0, 0, solutions)
```

Comment:
`flag` is not a self-documenting variable name.  What about `solved` instead?

Reply:
Or what about `if solutions:` instead.  Perhaps `run_maze()`should return `solutions` instead of taking it asan input parameter.

Reply:
> Or what about `if solutions:` instead. Perhaps `run_maze()`should return `solutions` instead of taking it as an input parameter.

Its recursive method. I can return solutions, but it will be same. solutions will be filled as input parameter. So, i think it will be same. I will fix other changes right now.